### ISSUE TITLE
checking for new updates and/or applying them

### ISSUE THREAD

**Issue by kir12** [TouhouFest/touhoufest-webapp#64]
due to the nature of caching with progressive web apps, it can be somewhat difficult to identify if new changes have been posted to a website and to get those changes in-place. For this reason, _some_ system of checking, reminding, and/or automatic updating should be in-place:
- create a text file with a version number and place it in a folder on the repository that will be packaged by `gh-pages`: e.g. `src`, `public`, etc.
- Create a class `UpdateCheck.tsx` that, when subject to some trigger, grabs the same text file directly from the GitHub repo (e.g. via GitHub API?)
  - Compare the text file that's cached and the text file that's pulled down. Display the results in a Bootstrap Modal with corresponding styling (e.g. green for good, red for needs updating)
  - Conditionally display instructions for pulling down the newest version if needed
- Add an option in the menu to manually run this modal, and automatically run it *once* on initial page load
  - Always display the modal if manually queued via the menu, but only display the modal on initial page load if a newer version is found

dependent on #62

**Comment by kir12** [comment]
follow-on optional augment:
- during the webpage deployment process, inscribe the running deployment's number as a text file via payload onto the page source
- have the github api read in the number of completed +running deployments and use that as a diff

**Comment by kir12** [comment]
#66 shelved and replaced with #62

**Comment by kir12** [comment]
notes:
- updating workbox dependencies was insufficient; pwa-builder will be tested as an alternate option (ideally in a separate branch)
- the need for a development deployment scheme is very clear here, and will probably be added as an additional target in a separate story

**Comment by kir12** [comment]
results:
- [vite pwa](https://vite-pwa-org.netlify.app/guide/pwa-minimal-requirements.html) was found to be an option that's closer to  vite so it was selected. reloads were found to be successfull, but it required deleting all site data in the process
- testing of subsequent change refreshes will need to take place gradually as features are merged
- a workwaround was found where site data could be deleted inside the browser. this should be added as a tip for users

**Comment by kir12** [comment]
update: the "install app" options on (at least) android browsers have disappeared not because PWAs arent working but because they've been renamed under both chrome and firefox. screenshots will need to be updated; this will be put together with the previously mentioned new story. in the meantime, new updates appear restored (tm) so this story will be closed, with provisions to re-open if necessary
